tests/pass/shims/x86/intrinsics-x86-vpclmulqdq.rs

```diff
@@ -0,0 +1,223 @@
+        unroll! {$target[2] = $op::<2>($source);}
+    };
+    ($target:ident[2] = $op:ident::<2>($source:ident);) => {
+        $target[1] = $op::<1>($source);
+        $target[0] = $op::<0>($source);
+    };
+    (assert_eq_m128i($op:ident::<4>($vec_res:ident),$lin_res:ident[4]);) => {
+        assert_eq_m128i($op::<3>($vec_res), $lin_res[3]);
+        assert_eq_m128i($op::<2>($vec_res), $lin_res[2]);
+        unroll! {assert_eq_m128i($op::<2>($vec_res),$lin_res[2]);}
+    };
+    (assert_eq_m128i($op:ident::<2>($vec_res:ident),$lin_res:ident[2]);) => {
+        assert_eq_m128i($op::<1>($vec_res), $lin_res[1]);
+        assert_eq_m128i($op::<0>($vec_res), $lin_res[0]);
+    };
+}
+
+// this function tests one of the possible 4 instances
+// with different inputs across lanes for the 512-bit version
+#[target_feature(enable = "vpclmulqdq,avx512f")]
+unsafe fn verify_512_helper(
+    linear: unsafe fn(__m128i, __m128i) -> __m128i,
+    vectorized: unsafe fn(__m512i, __m512i) -> __m512i,
+) {
+    let a = _mm512_set_epi64(
+        0xDCB4DB3657BF0B7D,
+        0x18DB0601068EDD9F,
+        0xB76B908233200DC5,
+        0xE478235FA8E22D5E,
+        0xAB05CFFA2621154C,
+        0x1171B47A186174C9,
+        0x8C6B6C0E7595CEC9,
+        0xBE3E7D4934E961BD,
+    );
+    let b = _mm512_set_epi64(
+        0x672F6F105A94CEA7,
+        0x8298B8FFCA5F829C,
+        0xA3927047B3FB61D8,
+        0x978093862CDE7187,
+        0xB1927AB22F31D0EC,
+        0xA9A5DA619BE4D7AF,
+        0xCA2590F56884FDC6,
+        0x19BE9F660038BDB5,
+    );
+
+    let mut a_decomp = [_mm_setzero_si128(); 4];
+    unroll! {a_decomp[4] = _mm512_extracti32x4_epi32::<4>(a);}
+    let mut b_decomp = [_mm_setzero_si128(); 4];
+    unroll! {b_decomp[4] = _mm512_extracti32x4_epi32::<4>(b);}
+
+    let r = vectorized(a, b);
+    let mut e_decomp = [_mm_setzero_si128(); 4];
+    for i in 0..4 {
+        e_decomp[i] = linear(a_decomp[i], b_decomp[i]);
+    }
+    unroll! {assert_eq_m128i(_mm512_extracti32x4_epi32::<4>(r),e_decomp[4]);}
+}
+
+// this function tests one of the possible 4 instances
+// with different inputs across lanes for the 256-bit version
+#[target_feature(enable = "vpclmulqdq")]
+unsafe fn verify_256_helper(
+    linear: unsafe fn(__m128i, __m128i) -> __m128i,
+    vectorized: unsafe fn(__m256i, __m256i) -> __m256i,
+) {
+    let a = _mm512_set_epi64(
+        0xDCB4DB3657BF0B7D,
+        0x18DB0601068EDD9F,
+        0xB76B908233200DC5,
+        0xE478235FA8E22D5E,
+        0xAB05CFFA2621154C,
+        0x1171B47A186174C9,
+        0x8C6B6C0E7595CEC9,
+        0xBE3E7D4934E961BD,
+    );
+    let b = _mm512_set_epi64(
+        0x672F6F105A94CEA7,
+        0x8298B8FFCA5F829C,
+        0xA3927047B3FB61D8,
+        0x978093862CDE7187,
+        0xB1927AB22F31D0EC,
+        0xA9A5DA619BE4D7AF,
+        0xCA2590F56884FDC6,
+        0x19BE9F660038BDB5,
+    );
+
+    let mut a_decomp = [_mm_setzero_si128(); 2];
+    unroll! {a_decomp[2] = _mm512_extracti32x4_epi32::<2>(a);}
```

Comment:
This function needs `avx512f`, so shouldn't we also enable that in the caller?

Reply:
I removed avx512 related code for the non-avx512 function.